File: BikeShop.Backend/BikeShop.Acts/BikeShop.Acts.Domain/Refit/Product.cs
using BikeShop.Payments.Domain.Entities;
using System.Text.Json.Serialization;

namespace BikeShop.Products.Domain.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string CatalogKey { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public string Barcode { get; set; } = string.Empty;
    public string? ManufacturerBarcode { get; set; }
    public int QuantityUnitId { get; set; }

    public decimal IncomePrice { get; set; }
    public decimal DealerPrice { get; set; }
    public decimal RetailPrice { get; set; }

    public int BrandId { get; set; }

    public string CheckStatus { get; set; } = string.Empty;
    public bool RetailVisibility { get; set; }
    public bool B2BVisibility { get; set; }
    
}

File: BikeShop.Frontend/webclient/src/entities/entities/Product.ts
export interface Product {
    id: number
    createdAt: string
    updatedAt: string
    enabled: boolean
    name: string
    catalogKey: string
    category: string
    barcode: string
    manufacturerBarcode: string
    quantityUnitId: number
    quantityUnitName: string
    incomePrice: number
    dealerPrice: number
    retailPrice: number
    brandId: number
    checkStatus: string
    retailVisibility: boolean
    b2BVisibility: boolean
}